### PR TITLE
Replace manual sform values with get_best_affine

### PR DESCRIPTION
**Pull request by frheault**
A simple switch to a much more robust Nibabel `get_best_affine()`

Manual getting the sform was often giving full zeros affine when they came from ITK/MITK/ANTs

### CODE REVIEW

**Comment by arokem** [comment]
Looks good to me. I don't know this function. Is it a relatively recent addition to nibabel? I don't think that would be a major concern as we are already requiring nibabel >3.0

File: dipy/io/utils.py [hunk 1]
```diff
@@ -268,9 +268,7 @@ def get_reference_info(reference):
 
     if is_nifti:
         affine = np.eye(4).astype(np.float32)
-        affine[0, 0:4] = header['srow_x']
-        affine[1, 0:4] = header['srow_y']
-        affine[2, 0:4] = header['srow_z']
+        affine = header.get_best_affine()
```
Comment (arokem) [inline dipy/io/utils.py]: Should we remove L270 as well?

**Comment by arokem** [comment]
This all looks good to me. Anyone else want to take a look?

File: dipy/io/tests/test_streamline.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ def io_tractogram(extension):
         nii_header = create_nifti_header(in_affine, in_dimensions,
                                          in_voxel_sizes)
         sft = StatefulTractogram(streamlines, nii_header, space=Space.RASMM)
-
+        print(nii_header.get_best_affine())
```
Comment (skoudoro) [inline dipy/io/tests/test_streamline.py]: Do we really need this print?

**Comment by arokem** [comment]
Thanks!

**Pull request merged**